src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -182,33 +181,47 @@ class HuiMapCard extends LitElement implements LovelaceCard {
   }
 
   protected shouldUpdate(changedProps: PropertyValues) {
+    if (changedProps.has("hass") || changedProps.has("_config")) {
+      const sourceEntities = this._getSourceEntities(
+        this.hass.states,
+        this._config
+      );
+      if (
+        !this._sourceEntities ||
+        sourceEntities.toString() !== this._sourceEntities.toString()
+      ) {
+        this._sourceEntities = sourceEntities;
```

Comment:
Should update should not have side effects, you can use `changedProps.get("hass")` to get the previous version of `hass`.

Reply:
Done. Removed "_sourceEntities" class member.